docs/src/main/asciidoc/getting-started-reactive.adoc

```diff
@@ -0,0 +1,854 @@
+quarkus.datasource.password=quarkus_test
+myapp.schema.create=true
+----
+
+The 3 first lines define the data source.
+Note database URL prefix, using `vertx-reactive` instead of the traditional `jdbc`.
+It indicates that this data source uses a reactive driver.
+
+The last line is going to be used in the application to indicate whether we insert a few items when the application gets initialized.
+
+Now, let's create our _entity_.
+Create the `org.acme.reactive.crud.Fruit` class with the following content:
+
+[source, java]
+----
+package org.acme.reactive.crud;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.pgclient.PgPool;
+import io.vertx.mutiny.sqlclient.Row;
+import io.vertx.mutiny.sqlclient.RowSet;
+import io.vertx.mutiny.sqlclient.Tuple;
+
+import java.util.stream.StreamSupport;
+
+public class Fruit {
+
+    public Long id;
+
+    public String name;
+
+    public Fruit() {
+        // default constructor.
+    }
+
+    public Fruit(String name) {
+        this.name = name;
+    }
+
+    public Fruit(Long id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+
+    public static Multi<Fruit> findAll(PgPool client) {
+        return client.query("SELECT id, name FROM fruits ORDER BY name ASC")
+                // Create a Multi from the set of rows:
+                .onItem().produceMulti(set -> Multi.createFrom().items(() -> StreamSupport.stream(set.spliterator(), false)))
+                // For each row create a fruit instance
+                .onItem().apply(Fruit::from);
+    }
+
+    public static Uni<Fruit> findById(PgPool client, Long id) {
+        return client.preparedQuery("SELECT id, name FROM fruits WHERE id = $1", Tuple.of(id))
+                .onItem().apply(RowSet::iterator)
+                .onItem().apply(iterator -> iterator.hasNext() ? from(iterator.next()) : null);
+    }
+
+    public Uni<Long> save(PgPool client) {
+        return client.preparedQuery("INSERT INTO fruits (name) VALUES ($1) RETURNING (id)", Tuple.of(name))
+                .onItem().apply(pgRowSet -> pgRowSet.iterator().next().getLong("id"));
+    }
+
+    public Uni<Boolean> update(PgPool client) {
+        return client.preparedQuery("UPDATE fruits SET name = $1 WHERE id = $2", Tuple.of(name, id))
+                .onItem().apply(pgRowSet -> pgRowSet.rowCount() == 1);
+    }
+
+    public static Uni<Boolean> delete(PgPool client, Long id) {
+        return client.preparedQuery("DELETE FROM fruits WHERE id = $1", Tuple.of(id))
+                .onItem().apply(pgRowSet -> pgRowSet.rowCount() == 1);
+    }
+
+    private static Fruit from(Row row) {
+        return new Fruit(row.getLong("id"), row.getString("name"));
+    }
+}
+----
+
+This _entity_ contains a few fields and methods to find, update, and delete rows from the database.
+These methods return either `Unis` or `Multis` as the produced items are emitted asynchronously when the database has retrieved the result.
```

Comment:
this is a bit weird `when the database has retrieved the result`. Is the database retrieving things?

Reply:
changing it to ` when the results have been retrieved.`